src/handlers/records-subscribe.ts

```diff
@@ -0,0 +1,225 @@
+      }
+    }
+    return filters;
+  }
+
+  /**
+   * Filters for only published records.
+   */
+  private static async filterPublishedRecords(
+    recordsSubscribe: RecordsSubscribe
+  ): Promise<Filter[]> {
+    const filter = RecordsSubscribeHandler.buildPublishedRecordsFilter(recordsSubscribe);
+    return [filter];
+  }
+
+  /**
+   * Creates a filter for all published records matching the subscribe
+   */
+  private static buildPublishedRecordsFilter(recordsSubscribe: RecordsSubscribe): Filter {
+    return {
+      ...Records.convertFilter(recordsSubscribe.message.descriptor.filter),
+      interface : DwnInterfaceName.Records,
+      method    : [ DwnMethodName.Write, DwnMethodName.Delete ],
+      published : true,
+    };
+  }
+
+  /**
+   * Creates a filter for unpublished records that are intended for the subscribe author (where `recipient` is the author).
+   */
+  private static buildUnpublishedRecordsForSubscribeAuthorFilter(recordsSubscribe: RecordsSubscribe): Filter {
+    // include records where recipient is subscribe author
+    return {
+      ...Records.convertFilter(recordsSubscribe.message.descriptor.filter),
+      interface : DwnInterfaceName.Records,
+      method    : [ DwnMethodName.Write, DwnMethodName.Delete ],
+      recipient : recordsSubscribe.author!,
+      published : false
+    };
+  }
+
+  /**
+   * Creates a filter for unpublished records that are within the specified protocol.
+   * Validation that `protocol` and other required protocol-related fields occurs before this method.
+   */
+  private static buildUnpublishedProtocolAuthorizedRecordsFilter(recordsSubscribe: RecordsSubscribe): Filter {
+    return {
+      ...Records.convertFilter(recordsSubscribe.message.descriptor.filter),
+      interface : DwnInterfaceName.Records,
+      method    : [ DwnMethodName.Write, DwnMethodName.Delete ],
+      published : false
+    };
+  }
+
+  /**
+   * Creates a filter for only unpublished records where the author is the same as the subscribe author.
+   */
+  private static buildUnpublishedRecordsBySubscribeAuthorFilter(recordsSubscribe: RecordsSubscribe): Filter {
+    // include records where author is the same as the subscribe author
+    return {
+      ...Records.convertFilter(recordsSubscribe.message.descriptor.filter),
+      author    : recordsSubscribe.author!,
+      interface : DwnInterfaceName.Records,
+      method    : [ DwnMethodName.Write, DwnMethodName.Delete ],
+      published : false
+    };
+  }
+
+  /**
+   * @param messageStore Used to check if the grant has been revoked.
+   */
+  public static async authorizeRecordsSubscribe(
+    tenant: string,
+    recordsSubscribe: RecordsSubscribe,
+    messageStore: MessageStore
+  ): Promise<void> {
+
+    if (Message.isSignedByDelegate(recordsSubscribe.message)) {
+      await recordsSubscribe.authorizeDelegate(messageStore);
+    }
+
+    // Only run protocol authz if message deliberately invokes it
```

Comment:
I am getting quite confused by this comment, going down the rabbit hole now.

Reply:
This is copy-pasta from `handlers/records-query.ts`.

There is definitely some room for refactoring some duplicate code between read/query/subscribe in this regard. Let me know if you'd like to chat about it.

Reply:
Yeah, I knew it came from query, but still was confused why the comment stated what it stated.

File this bug after digging, something that subscribe should probably take care of if not already:

https://github.com/TBD54566975/dwn-sdk-js/issues/669

Reply:
Good catch! Will discuss the implications within that specific issue.

Reply:
I have updated #669 to include subscribe when fixing.